R/shiny_prerendered.R

```diff
@@ -159,7 +158,26 @@ shiny_prerendered_html <- function(input_rmd, render_args) {
     dependencies <- append(dependencies, list(html_dependency_rsiframe()))
 
   # return html w/ dependencies
-  shinyHTML_with_deps(rendered_html, dependencies)
+  html_with_deps <- shinyHTML_with_deps(rendered_html, dependencies)
+
+  # add placeholder for additional dependencies at the end of the <head> element
```

Comment:
Adding the additional dependencies at the end of `<head>` feels like the right thing to do but unfortunately, it breaks previous behavior. Before this change the served document structure was (essentially):

```
shiny boilerplate:
|- head
|- body
   |- full prerendered document
      |- head
      |- body
```

so the dependencies added by shiny were in the first `<head>` tag and the document's dependencies appeared in the second `<head>` tag. When reconciled by the browser, this ends up being the same as

```
browser DOM
|- head
   |- shiny dependencies
   |- prerendered doc dependencies
```

In learnr, for example, this creates a situation where the scripts and dependencies in the tutorial's pandoc template appear _before_ the tutorial dependencies provided by learnr.

This may be the correct behavior and we can adjust in learnr, but I suspect there are other templates that take advantage of the implicit ordering of dependencies in the previous behavior.

Reply:
Followed in #2060 and fixed by #2064